engine/lib/entities.php

```diff
@@ -328,7 +329,8 @@ function can_write_to_container($user_guid = 0, $container_guid = 0, $type = 'al
 			array(
 				'container' => $container,
 				'user' => $user,
-				'subtype' => $subtype
+				'subtype' => $subtype,
+				'checking_owner' => ($checking === 'owner'),
```

Comment:
OK, and what do you think about just triggering a separate plugin hook entirely?